go-update/src/main/java/org/reactome/release/goupdate/GoTermsUpdater.java

```diff
@@ -137,57 +145,72 @@ else if (termStarted)
 		{
 			GoTermInstanceModifier goTermModifier;
 			GONamespace currentCategory = (GONamespace) goTermsFromFile.get(goID).get(GoUpdateConstants.NAMESPACE);
-			//String currentDefinition = (String) goTerms.get(goID).get(GoUpdateConstants.DEF);
 			// Now we need to process the Term that was just finished.
 			List<GKInstance> goInstances = allGoInstances.get(goID);
-			// First let's make sure the GO Term is not (pending) obsolete.
-			if ( !goTermsFromFile.get(goID).containsKey(GoUpdateConstants.IS_OBSOLETE) && !goTermsFromFile.get(goID).containsKey(GoUpdateConstants.PENDING_OBSOLETION))
+			// If there are no objects in allGoInstances whose accession is "goID", we may need to create it.
+			if (goInstances==null)
 			{
-				if (goInstances==null)
+				// BUT... only create a new instance if the data from the file is not Obsolete
+				if (!goTermsFromFile.get(goID).containsKey(GoUpdateConstants.IS_OBSOLETE) && !goTermsFromFile.get(goID).containsKey(GoUpdateConstants.PENDING_OBSOLETION))
 				{
 					// Create a new Instance if there is nothing in the current list of instances.
 					goTermModifier = new GoTermInstanceModifier(this.adaptor, this.instanceEdit);
-					newGoTermCount = createNewGOTerm(goTermsFromFile, goToECNumbers, newGoTermCount, goID, goTermModifier, currentCategory);
+					newGoTermCount++;
+					GKInstance newInst = createNewGOTerm(goTermsFromFile, goToECNumbers, goID, goTermModifier, currentCategory);
+					List<GKInstance> instList;
+					if (allGoInstances.containsKey(goID))
+					{
+						instList = allGoInstances.get(goID);
+					}
+					else
+					{
+						instList = new ArrayList<>(1);
+					}
+					instList.add(newInst);
+					allGoInstances.put(goID, instList);
 				}
-				else
+			}
+			else // update existing instance. Including Obsolete instances, because if they can't be deleted (for some reason) they should be updated so their name/def'n indicates their obsolescence.
+			{
+				// Try to update each goInstance that has the current GO ID.
+				for (GKInstance goInst : goInstances)
 				{
-					// Try to update each goInstance that has the current GO ID.
-					for (GKInstance goInst : goInstances)
+					// Compartment is a sub-class of GO_CellularComponent - but the GO namespaces don't seem to account for that,
+					// we we'll account for that here.
+					boolean categoryOK = goInst.getSchemClass().getName().equals(currentCategory.getReactomeName()) 
```

Comment:
Since boolean determination is subject to a variety of conditions, maybe there should be a method for it?